tgui/packages/tgui/interfaces/ModularShieldGen.tsx

```diff
@@ -0,0 +1,147 @@
+  current_strength: number;
+  max_regeneration: number;
+  current_regeneration: number;
+  max_radius: number;
+  current_radius: number;
+  active: Boolean;
+  recovering: Boolean;
+  exterior_only: Boolean;
+};
+
+export const ModularShieldGen = (props, context) => {
+  const { topLevel } = props;
+  const { act, data } = useBackend<modularShieldGeneratorData>(context);
+  const {
+    max_strength,
+    max_regeneration,
+    current_regeneration,
+    max_radius,
+    current_radius,
+    current_strength,
+    active,
+    exterior_only,
+    recovering,
+  } = data;
+
+  return (
+    <Window title="Modular Shield Generator" width={600} height={400}>
+      <Window.Content scrollable>
+        <Stack vertical fill>
+          <Stack.Item>
+            <Section
+              title={'Shield Strength'}
+              color={recovering ? 'red' : 'white'}>
+              <ProgressBar
+                title="Shield Strength"
+                value={current_strength}
+                minValue={0}
+                maxValue={max_strength}
+                height={'30px'}
+                ranges={{
+                  'good': [max_strength * 0.75, max_strength],
+                  'average': [max_strength * 0.25, max_strength * 0.75],
+                  'bad': [0, max_strength * 0.25],
+                }}>
+                {current_strength}/{max_strength}
+              </ProgressBar>
+            </Section>
+            <Section
+              horizontal
+              fill
+              height={'95px'}
+              title={'Regeneration and Radius'}>
+              <Section height={'20px'}>
+                <ProgressBar
+                  height={'20px'}
+                  title="Regeneration rate"
+                  value={current_regeneration}
+                  minValue={0}
+                  maxValue={max_regeneration}
+                  ranges={{
+                    'good': [max_regeneration * 0.75, max_regeneration],
+                    'average': [
+                      max_regeneration * 0.25,
+                      max_regeneration * 0.75,
+                    ],
+                    'bad': [0, max_regeneration * 0.25],
+                  }}>
+                  Regeneration {current_regeneration}/{max_regeneration}
+                </ProgressBar>
+              </Section>
+              <Section>
+                <ProgressBar
+                  height={'20px'}
+                  title="Shield radius"
+                  value={current_radius}
+                  minValue={0}
+                  maxValue={max_radius}
+                  ranges={{
+                    'good': [max_radius * 0.75, max_radius],
+                    'average': [max_radius * 0.25, max_radius * 0.75],
+                    'bad': [0, max_radius * 0.25],
+                  }}>
+                  Radius {current_radius}/{max_radius}
+                </ProgressBar>
+              </Section>
+              <Section title={'Settings'} bold={1}>
```

Comment:
no need to make 'Settings' an object like this. some other instances where this is the case also. title="Settings", title="Set Radius", etc